src/xrpld/app/misc/WasmHostFuncWrapper.cpp

```diff
@@ -67,23 +121,55 @@ getData(InstanceWrapper const* rt, int32_t src, int32_t ssz)
     if (src + ssz > mem.s)
         return Unexpected(HF_ERR_POINTER_OUT_OF_BOUNDS);
 
-    Bytes data(mem.p + src, mem.p + src + ssz);
+    Slice data(mem.p + src, ssz);
     return data;
 }
 
-static Expected<AccountID, int32_t>
-getDataAccount(InstanceWrapper const* rt, int32_t ptr, int32_t sz)
+template <>
+Expected<uint256, HostFunctionError>
+getData<uint256>(
+    InstanceWrapper const* rt,
+    wasm_val_vec_t const* params,
+    int32_t& i)
 {
-    auto const r = getData(rt, ptr, sz);
+    auto const r = getData<Slice>(rt, params, i);
+    i++;
```

Comment:
Ouch. We need to make UT protection from such  mistypes.
Please add additional unit test function (which should be present only when unit tests enabled) like` bool tesGetDataIndexIncrement() `to avoid to make `getData...` non static.  This function should made artificial call of all the `getData...` functions and just check that index is incremented correctly.  May be add additional template parameter for instance wrapper and create simple mock interface whith `getMem() `which will return `wmem` pointed to local vector. And add this call to `wasm_test` unit tests.